src/features/MainScreen/NotesList/NoteContextMenu/useNoteContextMenu.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback } from 'react';
+import { INote } from '@core/features/notes';
+import { useDynamicContextMenu } from '@hooks/useDynamicContextMenu';
+
+import { defaultNoteMenu, deletedNoteMenu } from './noteMenus';
+import {
+	ContextMenuOptions,
+	useNoteContextMenuCallback,
+} from './useNoteContextMenuCallback';
+
+/**
+ * Opens the context menu for a note.
+ * Automatically selects the appropriate menu based on the notes state
+ */
+export const useNoteContextMenu = ({ closeNote, updateNotes }: ContextMenuOptions) => {
+	const noteContextMenuCallback = useNoteContextMenuCallback({
+		closeNote,
+		updateNotes,
+	});
+	const openMenu = useDynamicContextMenu(noteContextMenuCallback, defaultNoteMenu);
+
+	return useCallback(
+		(note: INote, point: { x: number; y: number }) => {
+			const menu = note.isDeleted ? deletedNoteMenu : defaultNoteMenu;
```

Comment:
That's incorrect assumption that we have only 2 version of note menus.
We have uncountable number of menu variants. So we should build menu list in place.